### PR TITLE
feat(hierarchicalFacets): add add and remove operations on hierarchical facets

### PR DESCRIPTION
**Pull request by bobylito**

### CODE REVIEW

**Comment by redox** [comment]
Looks good :) Ping us when we need to review 👍

**Comment by bobylito** [comment]
You can review @redox :)

File: src/SearchParameters/index.js [hunk 1]
```diff
@@ -1280,6 +1280,41 @@ SearchParameters.prototype = {
...
+  /**
+   * Adds a refinement on a hierarchical facet.
+   * @param {string} facet the facet name
+   * @param {string} path the hierarchical facet path
+   * @return {SearchParameter} the new state
+   * @throws Error if the facet is not defined or if the facet is refined
+   */
+  addHierarchicalFacetRefinement: function(facet, path) {
+    if (this.isHierarchicalFacetRefined(facet)) {
+      throw new Error(facet + ' is already refined.');
```
Comment (redox) [inline src/SearchParameters/index.js]: We never throw Errors in such case before. Would it make sense to let such double-refinement happen?

Comment (redox) [inline src/SearchParameters/index.js]: But we may want to add a test to check if it's a hierarchical facet.

File: src/SearchParameters/index.js [hunk 2]
```diff
@@ -1280,6 +1280,41 @@ SearchParameters.prototype = {
...
+  },
+
+  /**
+   * Removes the refinement set on a hierarchical facet.
+   * @param {string} facet the facet name
+   * @return {SearchParameter} the new state
+   * @throws Error if the facet is not defined or if the facet is not refined
+   */
+  removeHierarchicalFacetRefinement: function(facet) {
+    if (!this.isHierarchicalFacetRefined(facet)) {
```
Comment (redox) [inline src/SearchParameters/index.js]: Check if it's a hierarchical facet first?

File: src/SearchParameters/index.js (hunk 1 above)
Comment (bobylito) [inline src/SearchParameters/index.js]: > But we may want to add a test to check if it's a hierarchical facet.

It's already done in `isHierarchicalFacetRefined`

> We never threw any errors in similar cases before. Would it make sense to let such double-refinement happen?

True. But it is how we ended up defining those methods in #250

File: src/SearchParameters/index.js (hunk 2 above)
Comment (bobylito) [inline src/SearchParameters/index.js]: See above comment :)

**Comment by vvo** [comment]
LGTM

**Pull request merged**